paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -435,18 +439,46 @@ internal class PaymentSheetViewModel @Inject internal constructor(
         ) : TransitionTarget()
     }
 
-    @Suppress("UNCHECKED_CAST")
     internal class Factory(
         private val applicationSupplier: () -> Application,
         private val starterArgsSupplier: () -> PaymentSheetContract.Args
-    ) : ViewModelProvider.Factory {
+    ) : ViewModelProvider.Factory, Injectable<Factory.FallbackInitializeParam> {
+        internal data class FallbackInitializeParam(
+            val application: Application,
+        )
+
+        @Inject
+        lateinit var subComponentBuilderProvider:
+            Provider<PaymentSheetViewModelSubcomponent.Builder>
+
         @Suppress("UNCHECKED_CAST")
         override fun <T : ViewModel?> create(modelClass: Class<T>): T {
-            return DaggerPaymentSheetViewModelComponent.builder()
-                .application(applicationSupplier())
-                .starterArgs(starterArgsSupplier())
-                .build()
-                .viewModel as T
+            val application = applicationSupplier()
+            val args = starterArgsSupplier()
+
+            val logger = Logger.getInstance(BuildConfig.DEBUG)
```

Comment:
It feels weird to me that the logger is not injected, it seems like it is in other places in the code base

Reply:
Yea, unfortunately at this point the injection graph is not initialized yet(that's why we need a logger to log whether injection is successful or not), one thing we can do is to pass `enableLogging` as a boolean to the Args like `Stripe3ds2TransactionViewModelFactory`, there is a couple of other places with similar changes, I can do that in a follow up